torch/utils/tensorboard/pytorch_graph.py

```diff
@@ -0,0 +1,274 @@
+import numpy as np
+import time
+import torch
+
+from collections import OrderedDict
+from distutils.version import LooseVersion
+
+from tensorboard.compat.proto.config_pb2 import RunMetadata
+from tensorboard.compat.proto.graph_pb2 import GraphDef
+from tensorboard.compat.proto.step_stats_pb2 import StepStats, DeviceStepStats, NodeExecStats, AllocatorMemoryUsed
+from tensorboard.compat.proto.versions_pb2 import VersionDef
+
+from .proto_graph import Node_proto
+
+
+methods_OP = ['attributeNames', 'hasMultipleOutputs', 'hasUses', 'inputs',
+              'kind', 'outputs', 'outputsSize', 'scopeName']
+methods_IO = ['node', 'offset', 'uniqueName']  # 'unique' <int> , 'type' <Tensor<class 'torch._C.Type'>>
+
+
+class Node_base(object):
+    def __init__(self, uniqueName=None, inputs=None, scope=None, tensorSize=None, op_type='UnSpecified', attributes=''):
+        self.uniqueName = uniqueName
+        self.inputs = inputs
+        self.tensorSize = tensorSize
+        self.kind = op_type
+        self.attributes = attributes
+        if scope is not None:
```

Comment:
This conditional seems a bit goofy. Wouldn't it be easier to just unconditionally set it? Now you have to check `hasattr(node_base, 'scope')` to test if it exists or not.

Reply:
Fixing.